cpp/src/arrow/array/builder_run_end.cc

```diff
@@ -289,6 +289,9 @@ Status RunEndEncodedBuilder::FinishInternal(std::shared_ptr<ArrayData>* out) {
   ARROW_ASSIGN_OR_RAISE(auto ree_array,
                         RunEndEncodedArray::Make(length_, run_ends_array, values_array));
   *out = std::move(ree_array->data());
+
+  UpdateDimensions(0, 0);
```

Comment:
```suggestion
  Reset();
```

Would this be ok in this scenerio?

Reply:
Oh nice; I missed this API. I just tried this and things work well on my end!